### PR TITLE
fix channels processing

### PR DESCRIPTION
**Pull request by DeadByElpy**

### CODE REVIEW

File: tests/commands.test.ts [hunk 1]
```diff
@@ -162,46 +162,44 @@ const uplinkCommands: TCommandList = [
...
+            header: '16 09',
+            body: '2f 97 aa 01 0c 83 01 08 0a',
+            lrc: 'd5'
         }
     },
     {
         constructor: uplink.DataHourMC,
         name: 'uplink command 0x17:DATA_HOUR_MC',
         parameters: {
+            seconds: 756648000,
```
[the remark this answers is not in the document]
Comment (DeadByElpy) [reply tests/commands.test.ts:182]: parameters reworked

File: src/CommandBinaryBuffer.ts [hunk 2]
```diff
@@ -114,53 +149,82 @@ class CommandBinaryBuffer extends BinaryBuffer {
...
 
     /**
      * Set array of channel indexes.
-     *
      */
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    setChannels ( channelList: Array<any> ) {
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-return
+    setChannels ( channelList: Array<IChannel> ) {
+        // sort to correct order of channels
```
[the remark this answers is not in the document]
Comment (DeadByElpy) [reply src/CommandBinaryBuffer.ts:191]: fixed

File: src/CommandBinaryBuffer.ts [hunk 3]
```diff
@@ -79,7 +114,7 @@ class CommandBinaryBuffer extends BinaryBuffer {
         while ( extended && this.offset <= this.data.byteLength ) {
             const byte = this.getUint8();
 
-            extended = byte & 0x80;
+            extended = byte & EXTENDED_BIT_MASK;
```
[the remark this answers is not in the document]
Comment (DeadByElpy) [reply src/CommandBinaryBuffer.ts:117]: renamed to 'isByteExtended' with boolean type

File: src/CommandBinaryBuffer.ts [hunk 4]
```diff
@@ -18,52 +18,66 @@ export interface IBatteryVoltage {
...
+
+export interface IChannelHourAbsoluteValue extends IChannelHours {
+    /**
+     * Channel pulse coefficient - IPK in bytes.
+     */
+    pulseCoefficient: number
+}
+
+export interface IChannelDays extends IChannel {
+    daysList: Array<number>
```
Comment (DarkPark) [inline src/CommandBinaryBuffer.ts:50]:
```suggestion
    dayList: Array<number>
```
?

Comment (DeadByElpy) [reply src/CommandBinaryBuffer.ts]: fixed

**Review by DarkPark — APPROVED** [review]

**Pull request merged**